PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/DemandFetcher.java

```diff
@@ -181,6 +181,11 @@ public void run() {
                         @MainThread
                         public void onDemandReady(final HashMap<String, String> demand, String auctionId) {
                             if (RequestRunnable.this.auctionId.equals(auctionId)) {
+                                if (requestParams.getNativeRequestParams() != null) {
+                                    if (demand.containsKey("hb_cache_id")) {
+                                        requestParams.getNativeRequestParams().setCacheId(demand.get("hb_cache_id"));
```

Comment:
I don't get this part. Why are you setting the cache id on the request?

Reply:
I am not setting it on the request, but using it to expose the cacheId through the NativeAdUnit instance.